src/extras/gizmo/gizmo.js

```diff
@@ -230,19 +236,35 @@ class Gizmo extends EventHandler {
     /**
      * Creates a new Gizmo object.
      *
-     * @param {AppBase} app - The application instance.
      * @param {CameraComponent} camera - The camera component.
-     * @param {Layer} layer - The render layer.
+     * @param {Layer} [layer] - The render layer.
      * @example
      * const gizmo = new pc.Gizmo(app, camera, layer);
      */
-    constructor(app, camera, layer) {
+    constructor(camera, layer) {
+        Debug.assert(camera instanceof CameraComponent, 'Incorrect parameters for Gizmos\'s constructor. Use new Gizmo(camera, layer)');
         super();
 
-        this._app = app;
-        this._device = app.graphicsDevice;
         this._camera = camera;
-        this._layer = layer;
+        this._app = camera.system.app;
+        this._device = this._app.graphicsDevice;
+
+        if (layer) {
+            this._layer = layer;
+        } else {
+            if (!gizmoLayer) {
```

Comment:
you should also check if the gizmo layer already exists (and the user didn't pass it in)

Reply:
I dont think this would be ideal since if the user created their own gizmo layer they could name it whatever they wanted. So either the user manages the gizmo layer entirely or gizmo manages the layers

Reply:
Also this would cause the layer that you created to be removed from the camera and scene if passed as an argument to the gizmo

Reply:
Can there be 2 gizmos in a scene?

Reply:
Yes the editor example uses all 3 of the transform gizmos

Reply:
layer name is unique .. we have a layer names map. So we need to make sure to not create a layer if one with the name exists.